packages/dds/tree/src/test/util/summaryBenchmarkSizes.ts

```diff
@@ -0,0 +1,142 @@
+    const tree = provider.trees[0];
+
+    const summarize = await provider.enableManualSummarization();
+    await summarize();
+    const summary = JSON.stringify(tree.getAttachSummary());
+    output.noEdits = IsoBuffer.from(summary).byteLength;
+
+    // summary sizes for inserting nodes.
+    output.insert1 = await getInsertsSummarySize(1);
+    output.insert10 = await getInsertsSummarySize(10);
+    output.insert100 = await getInsertsSummarySize(100);
+
+    // summary sizes for inserting then immediately removing nodes.
+    output.insertRemove1 = await getInsertsSummarySize(1, true);
+    output.insertRemove10 = await getInsertsSummarySize(10, true);
+    output.insertRemove100 = await getInsertsSummarySize(100, true);
+
+    return output;
+}
+
+/**
+ * Inserts a single node under the root of the tree with the given value.
+ */
+function setTestValue(
+    tree: ISharedTree,
+    value: TreeValue,
+    index: number): void {
+    // Apply an edit to the tree which inserts a node with a value
+    tree.runTransaction((forest, editor) => {
+        const writeCursor = singleTextCursor({ type: brand("TestValue"), value });
+        editor.insert({
+            parent: undefined,
+            parentField: detachedFieldAsKey(forest.rootField),
+            parentIndex: index,
+        }, writeCursor);
+
+        return TransactionResult.Apply;
+    });
+}
+
+function removeTestValue(tree: ISharedTree): void {
+    // Apply an edit to the tree which removes a node at index 0
+    tree.runTransaction((forest, editor) => {
+        editor.delete({
+            parent: undefined,
+            parentField: detachedFieldAsKey(forest.rootField),
+            parentIndex: 0,
+        }, 1);
+        return TransactionResult.Apply;
+    });
+}
+
+function setTestValues(
+    tree: ISharedTree,
+    numNodes: number): void {
+    for (let j = 0; j<numNodes; j++) {
+        setTestValue(
+            tree,
+            random.integer(
+                Number.MIN_SAFE_INTEGER,
+                Number.MAX_SAFE_INTEGER
+            ),
+            j
+        )
+    }
+}
+
+function setAndRemoveValues(
+    tree: ISharedTree,
+    numNodes: number): void {
+    for (let j = 0; j<numNodes; j++) {
+        setTestValue(
+            tree,
+            random.integer(
+                Number.MIN_SAFE_INTEGER,
+                Number.MAX_SAFE_INTEGER
+            ),
+            0
+        )
+        removeTestValue(
+            tree,
+        )
+    }
+}
+
+async function getInsertsSummarySize(numNodes: number, remove?: boolean): Promise<number> {
+    const provider = await TestTreeProvider.create(1);
+    const tree = provider.trees[0];
+
+    if (remove) {
+        setAndRemoveValues(tree, numNodes);
+    } else {
+        setTestValues(tree, numNodes);
+    }
+
+    const summarize = await provider.enableManualSummarization();
+    await summarize();
+    const summary = JSON.stringify(tree.getAttachSummary());
```

Comment:
In general, I don't think this is guaranteed to give you the size you are expecting. If the summary for example reused parts of a previous summary (perhaps one taken during the middle of the scenerio before you called enableManualSummarization) you maybe just get the size of a reference to those chunks.